### PR TITLE
GH-36969: [C++][R] Bump abseil-cpp version

### LINKED ISSUE #36969: [C++][R] abseil fails to build with gcc-13
### Describe the bug, including details regarding any error messages, version, and platform.

 e.g. https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=52112&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=d9b15392-e4ce-5e4c-0c8c-b69645229181&l=1975

```
/tmp/RtmpL1MTjz/file4a454972d15/absl_ep-prefix/src/absl_ep/absl/strings/internal/str_format/extension.cc:45:1: note: in expansion of macro ‘ABSL_INTERNAL_CONVERSION_CHARS_EXPAND_’
   45 | ABSL_INTERNAL_CONVERSION_CHARS_EXPAND_(ABSL_INTERNAL_CHAR_SET_CASE, )
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/RtmpL1MTjz/file4a454972d15/absl_ep-prefix/src/absl_ep/absl/strings/internal/str_format/extension.cc:44:13: error: ‘FormatConversionCharSet’ does not name a type; did you mean ‘FormatConversionCharInternal’?
   44 |   constexpr FormatConversionCharSet FormatConversionCharSetInternal::c;
      |             ^~~~~~~~~~~~~~~~~~~~~~~
```

### Component(s)

R

### PR DESCRIPTION
**Pull request by assignUser**
### Rationale for this change

Our current version of abseil-cpp fails to build with gcc 13 which is used on CRAN machines. 

### Are these changes tested?
CI
* Closes: #36969

### CODE REVIEW

**Comment by assignUser** [comment]
@github-actions crossbow submit -g cpp -g r

**Comment by h-vetinari** [comment]
Just a note that you need to be careful with abseil which C++ standard version you're using, as the ABI is affected by that choice (and you might e.g. run into missing symbols if one part is compiled with C++11 and one with C++17).

**Comment by assignUser** [comment]
@github-actions crossbow submit r-binary-packages test-fedora-35-cpp

**Comment by assignUser** [comment]
@github-actions crossbow submit r-binary-packages test-fedora-35-cpp

File: cpp/thirdparty/versions.txt [hunk 1]
```diff
@@ -23,8 +23,8 @@
 # `DEPENDENCIES` array (see the comment on top of the declaration for the
 # format).
 
-ARROW_ABSL_BUILD_VERSION=20211102.0
-ARROW_ABSL_BUILD_SHA256_CHECKSUM=dcf71b9cba8dc0ca9940c4b316a0c796be8fab42b070bb6b7cab62b48f0e66c4
+ARROW_ABSL_BUILD_VERSION=20230125.3
+ARROW_ABSL_BUILD_SHA256_CHECKSUM=5366d7e7fa7ba0d915014d387b66d0d002c03236448e1ba9ef98122c13b35c36
```
Comment (h-vetinari) [inline cpp/thirdparty/versions.txt:27]: JFYI, abseil very recently [released](https://github.com/abseil/abseil-cpp/releases/tag/20230802.0) a new LTS version.

Not sure if you're interested in upgrading this quickly, but presumably that'd mean you'd save half a year until the next abseil bump

**Pull request closed without merging**